src/foam/comics/v2/DAOSummaryView.js

```diff
@@ -107,18 +108,13 @@ foam.CLASS({
     },
     {
       name: 'delete',
-      confirmationRequired: true,
       code: function() {
-        this.config.dao.remove(this.data).then(o => {
-          this.finished.pub();
-          this.stack.back();
-        }, e => {
-          this.throwError.pub(e);
-          this.add(this.NotificationMessage.create({
-            message: e.message,
-            type: 'error'
-          }));
-        });
+        this.add(this.Popup.create().tag({
+          class: 'foam.u2.DeleteModal',
+          dao_: this.config.dao,
+          data: this,
```

Comment:
I would have set `data` to the object since views are usually of FObjects, not other views. 

I also think it would be more flexible to give `DeleteModal` a callback function that gets called on successful delete instead of passing the view in. Passing the view like you do now tightly couples `DeleteModal` to the `DAOSummaryView`, whereas passing a callback method keeps the two separate. This way other places can use `DeleteModal` as well and provide different callbacks. Eg:

```suggestion
          data: this.data,
          onDelete: () => {
            this.finished.pub();
            this.stack.back();
          }
```

Then in `DeleteModal` when you call `remove` you can just call `onDelete` if it succeeds and print the success message.

Reply:
yup like this!